front/app/hooks/useSomeFeatureFlags.ts

```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect } from 'react';
+import {
+  currentAppConfigurationStream,
+  IAppConfiguration,
+  TAppConfigurationSetting,
+} from 'services/appConfiguration';
+import { get } from 'lodash-es';
+
+type Parameters = {
+  names: TAppConfigurationSetting[];
+  onlyCheckAllowed?: boolean;
+};
+
+// copied and modified from front/app/hooks/useFeatureFlag.ts
+export default function useSomeFeatureFlags({
+  names,
+  onlyCheckAllowed,
+}: Parameters) {
+  const [tenantSettings, setTenantSettings] = useState<
+    | IAppConfiguration['data']['attributes']['settings']
+    | undefined
+    | null
+    | Error
+  >(undefined);
+
+  useEffect(() => {
+    const subscription = currentAppConfigurationStream().observable.subscribe(
+      (tenantSettings) =>
+        setTenantSettings(tenantSettings.data.attributes.settings)
+    );
+
+    return subscription.unsubscribe();
+  }, []);
+
+  return names.some(
+    (name) =>
+      get(tenantSettings, `${name}.allowed`) === true &&
+      (onlyCheckAllowed || get(tenantSettings, `${name}.enabled`) === true)
```

Comment:
I know it's just copied from useFeatureFlag, but we don't need lodash here (one dependency less). You could achieve the same here with:

`tenantSettings?.[name].allowed && (onlyCheckAllowed || tenantSettings?.[name].enabled`

You can also check isNilOrError for appConfig, return false is it is (seems like a sensible default if appConfig doesn't load?). Then you can leave out the optional chaining / `?.`

In general, I find the type checking of useFeatureFlag to be not so great, but let's keep the scope limited.

Reply:
Again, do you think it's worth changing `useFeatureFlag` too? Otherwise, I would keep it not perfect, but the same, to avoid confusion in the future (Why are these methods different though they do a similar thing? Maybe the logic differs too?)

Reply:
I'd lean towards 'yes' here, if it doesn't get you too deep into refactoring/testing. I think it's also going to be a one-line change, so then I'd go for it.

Reply:
(fixed)